src/cc_mm.c

```diff
@@ -34,10 +34,8 @@ _cc_alloc(size_t size, const char *name, int line)
 {
     void *p;
 
-    ASSERT(size != 0);
-
     p = malloc(size);
-    if (p == NULL) {
+    if (p == NULL && size != 0) {
```

Comment:
The return value is not `NULL` when size is 0, which is probably not the behavior we want.
I wrote the following test program:
```C
#include <stdio.h>
#include <stdlib.h>

int main()
{
    char *p = NULL;
    p = malloc(0);
    printf("%p\n", p);

    return 0;
}
```
and upon execution:
```sh
$ gcc mm.c
$ ./a.out
0x7f9c6ac031e0
```
check size at the top would be better just in general, as it is an exceptional condition/edge case.

```C
if (size == 0) {
    return NULL;
}

p = malloc(size);
...
```
    return